chain-factory/src/factory.rs

```diff
@@ -42,9 +61,39 @@ pub trait FactoryModule {
         let _ = self.all_deployed_contracts().swap_remove(&sc_address);
     }
 
+    #[only_owner]
+    #[endpoint(addContractsToMap)]
+    fn add_contracts_to_map(
+        &self,
+        contracts_list: MultiValueEncoded<Self::Api, ContractMap<Self::Api>>,
+    ) {
+        for contract in contracts_list {
+            require!(
+                self.blockchain().is_smart_contract(&contract.address),
+                "Given contract address is not a correct SC address"
+            );
+
+            require!(
+                self.is_required_contract_name(&contract.name),
+                "Given contract name is not a required SC"
+            );
```

Comment:
Instead of returning bool and checking, simply name the function `require_known_contract_name` and do the require there.